core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/date/DateRangeAggregationBuilder.java

```diff
@@ -285,7 +281,25 @@ protected DateRangeAggregatorFactory innerBuild(SearchContext context, ValuesSou
             AggregatorFactory<?> parent, Builder subFactoriesBuilder) throws IOException {
         // We need to call processRanges here so they are parsed and we know whether `now` has been used before we make
         // the decision of whether to cache the request
-        Range[] ranges = processRanges(context, config);
+        RangeAggregator.Range[] ranges = processRanges(range -> {
+            DocValueFormat parser = config.format();
+            assert parser != null;
+            double from = range.getFrom();
+            double to = range.getTo();
+            String fromAsString = range.getFromAsString();
+            String toAsString = range.getToAsString();
+            if (fromAsString != null) {
+                from = parser.parseDouble(fromAsString, false, context.getQueryShardContext()::nowInMillis);
+            } else if (Double.isFinite(from)) {
+                from = parser.parseDouble(Long.toString((long) from), false, context.getQueryShardContext()::nowInMillis);
```

Comment:
Although this will solve the case where the format is `epoch_millis` or `epoch_seconds`, does this not now mean that if you have a string based date format (e.g. `strict_date_optional_time`) you can no longer supply the from and to values as a long because the parser will throw an exception stating an invalid format?